src/managers/ThreadMemberManager.js

```diff
@@ -0,0 +1,96 @@
+'use strict';
+
+const BaseManager = require('./BaseManager');
+const ThreadMember = require('../structures/ThreadMember');
+const Collection = require('../util/Collection');
+
+/**
+ * Manages API methods for GuildMembers and stores their cache.
+ * @extends {BaseManager}
+ */
+class ThreadMemberManager extends BaseManager {
+  constructor(thread, iterable) {
+    super(thread.client, iterable, ThreadMember);
+    /**
+     * The thread this manager belongs to
+     * @type {ThreadChannel}
+     */
+    this.thread = thread;
+  }
+
+  /**
+   * The cache of this Manager
+   * @type {Collection<Snowflake, ThreadMember>}
+   * @name ThreadMemberManager#cache
+   */
+
+  add(data, cache = true) {
+    const existing = this.cache.get(data.user_id);
+    if (existing && existing._patch && cache) existing._patch(data);
```

Comment:
```suggestion
    if (cache) existing?._patch?.(data);
```
```suggestion
    if (cache && existing?._patch) existing._patch(data);
```
iirc _patch should be present on all structures so could even just cut out the method truthiness check

Reply:
This is probably a pattern from BaseManager that got copied into all managers, since ThreadMember obviously has _patch I'll remove the truthiness check here.